98_exercises.Rmd

```diff
@@ -291,24 +291,87 @@ See the OMA book for examples.
-
+The following exercise walks you through disassembling a MAE object in
+order to store it as multiple separate csv files.
+
+1. Import the microbiomeDataSets package
+2. Store the HintikkaXOData MAE into a variable named `mae`
+3. Which and how many experiments are contained by the MAE?
+4. Split the MAE into TreeSEs named `microbiota`, `metabolites` and `biomarkers`
+5. Which and how many assays are contained by each TreeSE?
+6. Store the assays of each TreeSE into a variable named `<TreeSE name>_assays`
+7. Write a csv file for each assays object with the `write.csv` function
+8. Besides experiments, MAEs also include a sampleData and a sampleMap, which
+   are accessible with `colData(mae)` and `sampleMap(mae)`, respectively.
+   Save also each of these two elements into a csv file.
+
+Well done! You just splitted a MAE into its components and stored them as csv files.
+[This script](https://github.com/JuliaTurkuDataScience/MicrobiomeAnalysis.jl/blob/main/src/assets/XO_preprocess.R)
+shows a possible approach. Next, we will try to reconstruct the same MAE
+from the files you created. Make sure you know their names and location!
+
+1. Read in the csv files containing assays with `read.csv` and save each
+   of them into a variable named `<assay name>_assays`
+2. Create one TreeSE from each assays object with the `TreeSummarizedExperiment`
+   function. Note that the function also requires the arguments for rowData
+   and colData, to which you can give the rownames and colnames of the
+   assays object, respectively
+3. Read in the sampleData and the sampleMap and store them into the
+   variables `sample_data` and `sample_map`, respectively
+4. Combine the TreeSEs, the sampleData and the sampleMap into a MAE named `mae`
+   with the `MultiAssayExperiment` function, where the first argument is an
+   `ExperimentList` of your TreeSEs, the second is colData and the third is
+   sampleMap
+5. Make sure that the MAE experiments are identical to the original TreeSEs.
+
+Good job! Now you are aware of how MAEs are built and we can proceed to some
+explorative exercises.
 
 ### Multi-omic data exploration
 
-1. Load experimental dataset from microbiomeDataSets (e.g., HintikkaXOData).
-2. Analyze correlations between experiments. (Taxa vs lipids, Taxa vs biomarkers, Lipids vs biomarkers)
-3. Agglomerate taxa data.
-4. Apply CLR to taxa data, apply log10 to lipids and biomarkers.
-5. Perform cross-correlation analyses and visualize results with heatmaps. (Use Spearman coefficients)
-6. Is there significant correlations? Interpret your results.
-
-Useful functions: pheatmap, ComplexHeatMap::Heatmap, ggplot, transformCounts, testExperimentCrossAssociation
+Here we learn some explorative methods and apply to the HintikkaXOData. If you
+solved the previous exercise, you can use the MAE you created and skip the first
+step.
+
```

Comment:
I would have the initial CSV saving and loading exercises _after_ the data manipulation exercises. This way students can first get motivation on why this thing can be useful, and then they may have more motivation to learn details on how to create it on your own.